### PR TITLE
ADCM-6077 Implement basic filters

### PR DESCRIPTION
**Pull request by Sealwing**

### CODE REVIEW

File: adcm_aio_client/core/filters.py [hunk 1]
```diff
@@ -10,5 +10,147 @@
...
+        if not allowed_filter:
+            message = f"Filter by {filter_.attr} is not allowed. Allowed: {', '.join(self._allowed)}"
+            raise InvalidFilterError(message)
+
+        if filter_.op not in allowed_filter.operations:
+            message = f"Operation {filter_.op} is not allowed. Allowed: {', '.join(sorted(allowed_filter.operations))}"
+            raise InvalidFilterError(message)
+
+        # we don't want to empty generator here
+        if isinstance(filter_.value, Generator):
```
Comment (DanBalalan) [inline adcm_aio_client/core/filters.py:93]: Isn't generators are tuples here already?  line 67 in to_query

File: adcm_aio_client/core/filters.py [hunk 2]
```diff
@@ -10,5 +10,143 @@
...
+
+    def _prepare_query_param_value(self: Self, value: SimplifiedValue) -> str:
+        if isinstance(value, tuple):
+            return ",".join(map(str, value))
+
+        return str(value)
+
+
+FilterByName = FilterBy("name", ALL_OPERATIONS, str)
+FilterByDisplayName = FilterBy("display_name", ALL_OPERATIONS, str)
```
Comment (a-alferov) [inline adcm_aio_client/core/filters.py:151]: What do you think if we don't give users the opportunity to use names with underscores, and force them to use names in the camelcase style?
I understand the disadvantages, you will have to use named arguments in methods not according to PEP8, for example `.filter(displayName__eq="service")`

Comment (Sealwing) [reply adcm_aio_client/core/filters.py:151]: I am against that, because field in objects are `display_name`, etc.

it's counter-intuitive

Comment (a-alferov) [reply adcm_aio_client/core/filters.py:151]: I agree

**Review by a-alferov — DISMISSED** [review]

**Review by a-alferov — APPROVED** [review]

**Review by Starovoitov — APPROVED** [review]

**Pull request merged**